cloudinit/sources/DataSourceEc2.py

```diff
@@ -170,9 +174,11 @@ def get_instance_id(self):
             if not getattr(self, 'identity', None):
                 # If re-using cached datasource, it's get_data run didn't
                 # setup self.identity. So we need to do that now.
+                headers = self._get_headers()
                 api_version = self.get_metadata_api_version()
                 self.identity = ec2.get_instance_identity(
-                    api_version, self.metadata_address).get('document', {})
+                    api_version, self.metadata_address, headers).get(
```

Comment:
This is a bug. We should drop the headers = self._get_headers()  above 
and this should pass both exception_cb and headers_cb instead of an invalid headers positional.


                self.identity = ec2.get_instance_identity(
                    api_version, self.metadata_address,
                    headers_cb=self._get_headers,
                    exception_cb=self._refresh_stale_aws_token_cb).get(
                        'document', {})

Since we'll definitely not have a token in the upgrade path, I expect that we'll need the _get_headers functionality which calls _refresh_api_token to refresh an absent api_token